### ISSUE TITLE
Init docker env before generate cluster config?

### ISSUE THREAD

**Issue by lingsamuel** [kubernetes/minikube#8883]
The function `generateClusterConfig` generates a new cluster config if `existing == nil`.
But it uses `config.DockerEnv` before initialize?

https://github.com/kubernetes/minikube/blob/master/cmd/minikube/cmd/start_flags.go#L294

https://github.com/kubernetes/minikube/blob/master/cmd/minikube/cmd/start_flags.go#L355

```go
// generateClusterConfig generate a config.ClusterConfig based on flags or existing cluster config
func generateClusterConfig(cmd *cobra.Command, existing *config.ClusterConfig, k8sVersion string, drvName string) (config.ClusterConfig, config.Node, error) {
	var cc config.ClusterConfig
	if existing != nil {
		cc = updateExistingConfigFromFlags(cmd, existing)
	} else {
                // .....
		cc = config.ClusterConfig{
                        // ....
			DockerEnv:               config.DockerEnv,
                        // ....
			},
		}
                // ......
	}

        // .....
	// Feed Docker our host proxy environment by default, so that it can pull images
	// doing this for both new config and existing, in case proxy changed since previous start
	if _, ok := r.(*cruntime.Docker); ok {
		proxy.SetDockerEnv()
	}
        // .....
```

If add some logs to test `TestGenerateCfgFromFlagsHTTPProxyHandling` in `cmd/minikube/cmd/start_test.go`, the generated config has wrong docker env (I think the test is wrong by the way).

**Comment by priyawadhwa** [comment]
Hey @lingsamuel so it seems like the default value for `config.DockerEnv` is `nil`:

https://github.com/kubernetes/minikube/blob/63bfa7a444693903e87b63d7ee2a3dc739ee2dde/cmd/minikube/cmd/start_flags.go#L207

which seems correct to me for a brand new cluster. Do you still think there's a bug?

**Comment by lingsamuel** [comment]
My fault, I forget to close this. The problem I encountered is a bug in the test, which I already fixed it in #8885
